packages/template-retail-react-app/app/pages/checkout/index.test.js

```diff
@@ -320,136 +290,8 @@ test.skip('Can proceed through checkout steps as guest', async () => {
-        }),
-
-        // mock add shipping method
-        rest.put('*/shipments/me/shipping_method', (req, res, ctx) => {
-            currentBasket.shipments[0].shipping_method =
-                mockShippingMethods.applicable_shipping_methods[0]
-            return res(ctx.json(currentBasket))
-        }),
-
-        // mock add payment instrument
-        rest.post('*/baskets/:basketId/payment_instruments', (req, res, ctx) => {
-            currentBasket.payment_instruments = [
-                {
-                    amount: 0,
-                    payment_card: {
-                        cardType: 'Master Card',
-                        creditCardExpired: false,
-                        expirationMonth: 1,
-                        expirationYear: 2030,
-                        holder: 'Test McTester',
-                        maskedNumber: '************5454',
-                        numberLastDigits: '5454',
-                        validFromMonth: 1,
-                        validFromYear: 2020
-                    },
-                    payment_instrument_id: 'testcard1',
-                    payment_method_id: 'CREDIT_CARD',
-                    _type: 'order_payment_instrument'
-                }
-            ]
-            return res(ctx.json(currentBasket))
-        }),
-
-        // mock update address
-        rest.patch('*/addresses/savedaddress1', (req, res, ctx) => {
-            return res(ctx.json(mockedRegisteredCustomer.addresses[0]))
-        }),
-
-        // mock place order
-        rest.post('*/orders', (req, res, ctx) => {
-            currentBasket = {
-                ...ocapiOrderResponse,
-                customer_info: {...ocapiOrderResponse.customer_info, email: 'customer@test.com'}
-            }
-            return res(ctx.json(currentBasket))
-        })
-    )
-
-    // Set the initial browser router path and render our component tree.
-    window.history.pushState({}, 'Checkout', createPathWithDefaults('/checkout'))
-    renderWithProviders(<WrappedCheckout history={history} />, {
-        wrapperProps: {siteAlias: 'uk', locale: {id: 'en-GB'}, appConfig: mockConfig.app}
-    })
-
-    // Switch to login
-    const haveAccountButton = await screen.findByText(/already have an account/i)
-    user.click(haveAccountButton)
-
-    // Wait for checkout to load and display first step
-    const loginBtn = await screen.findByText(/log in/i)
-
-    // Provide customer email and submit
-    const emailInput = screen.getByLabelText('Email')
-    const pwInput = screen.getByLabelText('Password')
-    user.type(emailInput, 'customer@test.com')
-    user.type(pwInput, 'Password!1')
-    user.click(loginBtn)
-
-    // Wait for next step to render
-    await waitFor(() =>
-        expect(screen.getByTestId('sf-toggle-card-step-1-content')).not.toBeEmptyDOMElement()
-    )
+test('Can proceed through checkout as registered customer', async () => {
+    await logInDuringCheckout()
```

Comment:
A few tests had duplicate logic to log in during the checkout flow. So here I extracted it out into its own function.